test/nd.jl

```diff
@@ -92,9 +92,18 @@ Base.zero(::Type{WrappedFloat}) = WrappedFloat(0.0)
         @test all(x->x==0, imgf[first(inds):-2]) && all(x->x==0, imgf[2:last(inds)])
     end
 
-    # Input with NaN
-    @test isnan.(imfilter([NaN;1:100],centered(ones(31)))) == ((0:100) .<= 15)
-
+    # Non-finite input
+    for x in (NaN, Inf, -Inf)
+        v = rand(100)
+        i = rand(eachindex(v))
+        w = [j == i ? x : v[j] for j in eachindex(v)]
+        kern = centered(ones(31))
+        vf = imfilter(v, kern)
+        wf = imfilter(w, kern)
```

Comment:
To make sure that the spirit of this test is retained against future code changes (e.g., what if someone changes `L > 30` to `L > 45`?), we might want to explicitly test `ImageFiltering.filter_algorithm(w, w, kern) == FFT()`.

I acknowledge that I didn't do this carefully in the original test, so the lack isn't your fault.

Reply:
Actually, I believe the original test here is mine too. Anyway, I added the algorithm check.